MapboxCoreNavigation/RouteStep.swift

```diff
@@ -38,4 +39,46 @@ extension RouteStep {
     open var lastInstruction: SpokenInstruction? {
         return instructionsSpokenAlongStep?.last
     }
+    
+    /**
+     Returns true if the current route step contains a tunnel.
+     */
+    var containsTunnel: Bool {
+        guard let intersections = intersections else { return false }
+        for intersection in intersections {
+            if intersection.outletRoadClasses?.contains(.tunnel) == true {
+                return true
+            }
+        }
+        return false
+    }
+    
+    /**
+     Returns a tunnel slice for the current route step coordinates
+     */
+    var tunnelSlice: Polyline? {
+        guard let coordinates = coordinates, let tunnelIntersection = tunnelIntersection else { return nil }
+        return Polyline(coordinates).sliced(from: tunnelIntersection.entry.location, to: tunnelIntersection.exit.location)
+    }
+    
+    /**
+     Returns the distance of a tunnel slice for the current route step
+     */
+    var tunnelDistance: CLLocationDistance? {
+        guard let tunnelIntersection = tunnelIntersection else { return nil }
+        return tunnelIntersection.entry.location.distance(to: tunnelIntersection.exit.location)
+    }
+    
+    /**
+     Returns the entry and exit intersections of the tunnel on the current route step
+     */
+    var tunnelIntersection: (entry: Intersection, exit: Intersection)? {
```

Comment:
`tunnelIntersection ` denotes a single intersection, while in reality, we're calculating the entry and exit of a tunnel here. Could you come up with a more descriptive variable name for this variable?